docs/enhancements/4-nodes-cluster-deployment.md

```diff
@@ -0,0 +1,154 @@
+enclosure that only had 4 nodes. The goal was to use all of the blades rather than just 3 of them.
+
+For now, this is achieved with a 2 step deployment, first 3 masters and then add a worker node. It
+would be more time efficient as well as easier and more reliable to be able to do this deployment in
+one step, day0.
+
+# Motivation
+
+Assisted service has built-in validations that prevent the deployment of a 4-nodes cluster. This is
+to prevent unhealthy clusters to be deployed. A 4-nodes cluster ends up in an unhealthy state
+because openshift-installer doesn't take schedulable masters into consideration when there are
+workers being deployed. The problem with schedulable masters not being counted as workers causes the
+installer to chose an `InfrastructureTopology` that is SingleReplica. This means that infrastructure
+workloads will run as single replicas, only on workers. This setting cannot be changed once the
+cluster is deployed.
+
+The above was brought up to the OpenShift team as [an enhancement](https://github.com/openshift/enhancements/pull/1057/). The feedback so far is that
+it's OK for users (or other installers) to overwrite the openshift manifests to achieve the
+deployment topologies required. The conversation is still in progress as far as understanding if
+some changes could be done to the openshift installer to improve the user experience here but we
+were encourage to pursue manifests' overwrite as a way to provide this through assisted service.
+
+### Goals
+
+- Allow for a 4-nodes cluster to be created using the KubeAPI flow
+
+### Non-Goals
+
+- Implement logic that will decide, for the user, what the best architecture is. Having a 3+1
+cluster will require that masters are explicitly set to schedulable and the manifests to patch the
+Infrastructure CR are provided.
+
+- Expose this through SaaS at this stage. This would require changes in the UI and other validations
+that may not worth the effort at this stage.
+
+## Proposal
+
+The biggest changes anticipated for this enhancement are:
+
+- Expose the manifests API through the InfraEnv CRD so that custom manifests can be added through
+the ZTP flow.
+
+- Expose the schedulable mastes through the AgentClusterInstall to allow for explicitly creating
+schedulable masters.
+
+- Modify the `enoughMastersAndWorkers` validation so that it allows 1 worker when masters are
+scheulable. [PoC here](https://github.com/flaper87/assisted-service/commit/f07fc6589e4f2d316266ac5533d76bd2faf471d7)
+and [here](https://github.com/flaper87/assisted-service/commit/bb2d82eaacaee7790d2f18711e2ba3a442a4967d)
+
+- Modify Assisted Service's manifests logic to allow for partial manifests to be provided. Openshift
+installer applies the provided manifests in sequence. If 2 manifests for the same CR are provided,
+then the one applied last will be the one used. This is a problem when the user only wants to modify
```

Comment:
Could you elaborate on what you mean exactly by "applied last"? Usually manifest apply ordering is done with `xx_` numerical prefixes to the manifest file names, maybe we can have a similar scheme here as well?